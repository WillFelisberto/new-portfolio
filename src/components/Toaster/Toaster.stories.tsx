import { Meta, StoryObj } from '@storybook/react';
import Toaster from '.';
import toast from 'react-hot-toast';

const meta: Meta<typeof Toaster> = {
  title: 'Components/Toaster',
  component: Toaster,
  parameters: {
    layout: 'fullscreen',
  },
};

export default meta;

type Story = StoryObj<typeof Toaster>;

export const Default: Story = {
  render: () => {
    // Função para disparar notificações de exemplo
    const triggerToast = () => {
      toast.success('This is a success message!');
      toast.error('This is an error message!');
    };

    return (
      <>
        <Toaster />
        <div style={{ textAlign: 'center', marginTop: '2rem' }}>
          <button
            onClick={triggerToast}
            style={{
              padding: '10px 20px',
              background: '#0070f3',
              color: '#fff',
              border: 'none',
              borderRadius: '5px',
              cursor: 'pointer',
            }}
          >
            Trigger Toast
          </button>
        </div>
      </>
    );
  },
};
