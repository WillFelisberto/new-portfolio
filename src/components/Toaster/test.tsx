import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Toaster from '.';
import { toast } from 'react-hot-toast';

// Mock de react-hot-toast
jest.mock('react-hot-toast', () => ({
  toast: {
    success: jest.fn(),
    error: jest.fn(),
  },
  Toaster: jest.fn(() => <div data-testid="toaster-provider" />),
}));

describe('Toaster component', () => {
  it('renders the ToasterProvider', () => {
    const { container } = render(<Toaster />);
    const toaster = screen.getByTestId('toaster-provider');
    expect(toaster).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('displays a success toast when triggered', () => {
    render(<Toaster />);
    toast.success('This is a success message!');
    expect(toast.success).toHaveBeenCalledWith('This is a success message!');
  });

  it('displays an error toast when triggered', () => {
    render(<Toaster />);
    toast.error('This is an error message!');
    expect(toast.error).toHaveBeenCalledWith('This is an error message!');
  });

  it('triggers a success and error toast when button is clicked', () => {
    const triggerToast = () => {
      toast.success('This is a success message!');
      toast.error('This is an error message!');
    };

    const ButtonComponent = () => (
      <>
        <Toaster />
        <button data-testid="trigger-toast" onClick={triggerToast}>
          Trigger Toast
        </button>
      </>
    );

    render(<ButtonComponent />);

    const button = screen.getByTestId('trigger-toast');
    fireEvent.click(button);

    expect(toast.success).toHaveBeenCalledWith('This is a success message!');
    expect(toast.error).toHaveBeenCalledWith('This is an error message!');
  });
});
