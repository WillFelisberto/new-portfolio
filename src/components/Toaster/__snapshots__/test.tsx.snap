// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toaster component renders the ToasterProvider 1`] = `
<div
  data-testid="toaster-provider"
/>
`;
